subprojects/parseq/src/test-utils/java/com/linkedin/parseq/ParSeqUnitTestHelper.java

```diff
@@ -325,6 +325,10 @@ public int countTasks(Trace trace) {
     public void onPlanCompleted(PlanContext planContext) {
       CountDownLatch latch = _taskDoneLatch.computeIfAbsent(planContext.getRootTask(), key -> new CountDownLatch(1));
       latch.countDown();
+      
+      if (latch.getCount() == 0L) {
```

Comment:
This fix may not work. This call is called, but some tests may need to call `await` .

Reply:
from what I understand, this onPlanCompleted is called by the engine when it is done.
Are you suggesting that even in await, if latch.await returns false- that is the work wasn't done, we forcibly remove the latch?

Reply:
Capture what we have discussed offline here. We need to probably handle two cases to make sure that `runAndWaitForPlanComplete` API in ParSeqUnitTestHelper not break.
1. TaskDoneListener.await is called before TaskDoneListener.onPlanCompleted.
2. TaskDoneListener.await is called after TaskDoneListener.onPlanCompleted.